Source/Core/Common/VR/VRRenderer.cpp

```diff
@@ -0,0 +1,507 @@
+        (XrViewConfigurationType*)malloc(viewport_config_count * sizeof(XrViewConfigurationType));
+
+    OXR(xrEnumerateViewConfigurations(engine->app_state.instance, engine->app_state.system_id,
+                                      viewport_config_count, &viewport_config_count,
+                                      viewport_configs));
+
+    ALOGV("Available Viewport Configuration Types: %d", viewport_config_count);
+
+    for (uint32_t i = 0; i < viewport_config_count; i++)
+    {
+      const XrViewConfigurationType viewport_config_type = viewport_configs[i];
+
+      ALOGV("Viewport configuration type %d : %s", viewport_config_type,
+            viewport_config_type == XR_VIEW_CONFIGURATION_TYPE_PRIMARY_STEREO ? "Selected" : "");
+
+      XrViewConfigurationProperties viewport_config;
+      viewport_config.type = XR_TYPE_VIEW_CONFIGURATION_PROPERTIES;
+      OXR(xrGetViewConfigurationProperties(engine->app_state.instance, engine->app_state.system_id,
+                                           viewport_config_type, &viewport_config));
+      ALOGV("FovMutable=%s ConfigurationType %d", viewport_config.fovMutable ? "true" : "false",
+            viewport_config.viewConfigurationType);
+
+      uint32_t view_count;
+      OXR(xrEnumerateViewConfigurationViews(engine->app_state.instance, engine->app_state.system_id,
+                                            viewport_config_type, 0, &view_count, NULL));
+
+      if (view_count > 0)
+      {
+        XrViewConfigurationView* elements =
+            (XrViewConfigurationView*)malloc(view_count * sizeof(XrViewConfigurationView));
+
+        for (uint32_t e = 0; e < view_count; e++)
+        {
+          elements[e].type = XR_TYPE_VIEW_CONFIGURATION_VIEW;
+          elements[e].next = NULL;
+        }
+
+        OXR(xrEnumerateViewConfigurationViews(engine->app_state.instance,
+                                              engine->app_state.system_id, viewport_config_type,
+                                              view_count, &view_count, elements));
+
+        // Cache the view config properties for the selected config type.
+        if (viewport_config_type == XR_VIEW_CONFIGURATION_TYPE_PRIMARY_STEREO)
+        {
+          assert(view_count == MaxNumEyes);
+          for (uint32_t e = 0; e < view_count; e++)
+          {
+            engine->app_state.view_config[e] = elements[e];
+          }
+        }
+
+        free(elements);
+      }
+      else
+      {
+        ALOGE("Empty viewport configuration type: %d", view_count);
+      }
+    }
+
+    free(viewport_configs);
+
+    *pWidth = width = engine->app_state.view_config[0].recommendedImageRectWidth;
+    *pHeight = height = engine->app_state.view_config[0].recommendedImageRectHeight;
+  }
+  else
+  {
+    // use cached values
+    *pWidth = width;
+    *pHeight = height;
+  }
+}
+
+void Renderer::Init(engine_t* engine, bool multiview)
+{
+  if (initialized)
+  {
+    Destroy(engine);
```

Comment:
What is this protecting against?

Reply:
This makes destroyes previous instance of the renderer if Init is called multiple times. Init of the renderer is called everytime the display goes on.